### PR TITLE
add monasca support

### PR DESCRIPTION
**Pull request by sjamgade**
Pull request for master branch

### CODE REVIEW

File: scripts/openstack-quickstart-demosetup [hunk 1]
```diff
@@ -148,6 +174,16 @@ if [ "$DB" = "postgresql" ] ; then
         sudo -u postgres createdb -O $DBNAME $DBNAME || true
     done
     sudo -u postgres createuser -s root || :
+
+  # Monasca needs MySQL either way, so create a MySQL database for its use
+  echo | mysql -u root || pwquery=-p
+  echo "
+  set global character_set_server=latin1;
```
Comment (toabctl) [inline scripts/openstack-quickstart-demosetup]: Isn't that the same we do already in the loop starting from line#143 ?

File: scripts/openstack-quickstart-demosetup [hunk 2]
```diff
@@ -816,6 +855,91 @@ if [ "x$with_gnocchi" = "xyes" ]; then
     #start_and_enable_service openstack-gnocchi-statsd
 fi
 
+#-----------------------------------------
+# setup Monasca
+#-----------------------------------------
+
+if [ "x$with_monasca" = "xyes" ]; then
+# prerequisite services
```
Comment (toabctl) [inline scripts/openstack-quickstart-demosetup]: indent missing

**Comment by toabctl** [comment]
This produces now a working monasca setup (still some tempest tests failing but most passing). It depends on 2 patches for the monasca-installer package:
- https://gitlab.com/monasca-installer/ansible-monasca-log-api/merge_requests/1/
- https://gitlab.com/monasca-installer/ansible-memcached/merge_requests/1

File: scripts/openstack-quickstart-demosetup [hunk 3]
```diff
@@ -69,6 +69,28 @@ if [ "x$with_gnocchi" = "xyes" ]; then
                      python-gnocchiclient
 fi
 
+if [ "x$with_monasca" = "xyes" ]; then
+    install_packages openstack-monasca-log-agent openstack-monasca-agent \
```
Comment (dirkmueller) [inline scripts/openstack-quickstart-demosetup]: ideally that would be an openstack-monitoring pattern..

Comment (toabctl) [inline scripts/openstack-quickstart-demosetup]: Hm. you mean a client and a server pattern for monitoring?

**Pull request merged**